diff --git a/dist/index.js b/dist/index.js
index 78bd0669875fd9c152a4d77763be110dfa87265a..e59013fa5e576c76903f4fb603123cc9dddbe892 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -66,13 +66,17 @@ var CloudflareDeployer = class extends Deployer {
   getEntry() {
     return `
     import '#polyfills';
-    import { mastra } from '#mastra';
-    import { createHonoServer } from '#server';
     import { evaluate } from '@mastra/core/eval';
     import { AvailableHooks, registerHook } from '@mastra/core/hooks';
     import { TABLE_EVALS } from '@mastra/core/storage';
     import { checkEvalStorageFields } from '@mastra/core/utils';
 
+    export default {
+      fetch: async (request, env, context) => {
+
+    const { createHonoServer } = await import('#server');
+    const { mastra } = await import('#mastra');
+
     registerHook(AvailableHooks.ON_GENERATION, ({ input, output, metric, runId, agentName, instructions }) => {
       evaluate({
         agentName,
@@ -111,8 +115,6 @@ var CloudflareDeployer = class extends Deployer {
       }
     });
 
-    export default {
-      fetch: async (request, env, context) => {
         const app = await createHonoServer(mastra)
         return app.fetch(request, env, context);
       }
